src/expectEvent.js

```diff
@@ -108,10 +108,17 @@ async function notInTransaction (txHash, emitter, eventName) {
 // the same form truffle-contract uses on its receipts
 function decodeLogs (logs, emitter, eventName) {
   let abi;
+  let address;
   if (isWeb3Contract(emitter)) {
     abi = emitter.options.jsonInterface;
+    address = emitter.options.address;
   } else if (isTruffleContract(emitter)) {
     abi = emitter.abi;
+    try {
```

Comment:
What exception is this `try` block meant to catch? Wouldn't `emitter.address` just be `undefined` and result in `address = undefined`?

Reply:
Yes, if the contract emitter address is null it throws, therefore it needs this try catch, maybe could be replaced with a ifelse.

Reply:
Ah, this happens when the expectEvent function is used with a "contract artifact class" (no deploy address). This was added to ensure backwards compatibility, so tests don't need to be changed to use an artifact with a deployed address, this is only needed for when there is conflicting indexed events.